Adds Borg Hypo Chem Descriptions

V2LenKagamine opened the pull request:
## About The Pull Request
Adds a short description of what each Chemical in the Hypospray does.

## Why It's Good For The Game
 Sometimes I just don't want to have to look at the wiki every 10 seconds to make sure im healing this dying patient for a fuckshit of brute instead of burn. This also can remove accidental "Human Harm" from borgs injecting the wrong chemical into their patient.
## Changelog
:cl:
add: Added Borg Hypospray chem descriptions
/:cl:

V2LenKagamine committed:
Adds Borg Hypo Chem Descriptions

Adds descriptions to what each chemical does in the borg hypo spray

V2LenKagamine commented:
Big thanks to "TheFinalPotato" for effectively coding the whole thing by walking me through it.

ATH1909 commented on code/modules/reagents/reagent_containers/borghydro.dm:
```diff
@@ -209,15 +203,14 @@ Borg Shaker
+	/datum/reagent/consumable/space_up, /datum/reagent/consumable/tea, /datum/reagent/consumable/tomatojuice,
+	/datum/reagent/consumable/tonic, /datum/reagent/water,
 	/datum/reagent/consumable/ethanol/ale, /datum/reagent/consumable/ethanol/applejack, /datum/reagent/consumable/ethanol/beer,
-	/datum/reagent/consumable/ethanol/champagne, /datum/reagent/consumable/ethanol/cognac, /datum/reagent/consumable/ethanol/creme_de_menthe,
```
undocumented change

V2LenKagamine replied:
that shoudlnt be a change, ill fix that

zxaber commented:
I'm confused, I did almost this exact same thing in #46140. Did that break or something since then?

Also you seem to have a bunch of unrelated changes in the borg shaker.

V2LenKagamine committed:
Re adds Accidentally removed shaker code

Thanks for pointing that out

V2LenKagamine committed:
Fixes code

Removes minor coding errors causing compile fail. Also Tested and working.

LemonInTheDark commented on code/modules/reagents/reagent_containers/borghydro.dm:
```diff
@@ -1,4 +1,5 @@
 #define C2NAMEREAGENT	"[initial(reagent.name)] (Has Side-Effects)"
```
This isn't used anywhere, you can probably remove it.

V2LenKagamine replied:
Removed

LemonInTheDark commented on code/modules/reagents/reagent_containers/borghydro.dm:
```diff
@@ -1,4 +1,5 @@
 #define C2NAMEREAGENT	"[initial(reagent.name)] (Has Side-Effects)"
+#define CHEMEFFECT "[initial(reagent.name)][reagent_desc[reagent]] [initial(reagent.harmful) ? "(Has side effects)" : ""]"
```
As it currently stands this doesn't check to see if the reagent is actually in reagent_desc, so it will runtime if we try and get a reagent that doesn't have an entry.

To check for an entry do something like this:
```suggestion
#define CHEMEFFECT "[initial(reagent.name)][(reagent in reagent_desc) ? reagent_desc[reagent] : ""] [initial(reagent.harmful) ? "(Has side effects)" : ""]"
```
Oh and you may want to add some spaces between the ][s so the textisn'tdisplayedlikethis.

LemonInTheDark commented:
Ya did good

V2LenKagamine committed:
Adds Reagent Desc check

Adds check to make sure reagent is in reagent desc. to avoid runtime errors per Contributor suggestion

Co-Authored-By: LemonInTheDark <58055496+LemonInTheDark@users.noreply.github.com>

V2LenKagamine committed:
Removes Unessicary code

Removes 2 lines of now obsolete code.

Tested and working as intended.

spookydonut commented on code/modules/reagents/reagent_containers/borghydro.dm:
```diff
@@ -32,7 +32,7 @@ Borg Hypospray
 								//Used as list for input() in shakers.
 	var/list/reagent_names = list()
-
+	var/list/reagent_desc = list(/datum/reagent/medicine/C2/convermol = " (Heals Oxygen)", /datum/reagent/medicine/C2/libital = " (Heals Brute)", /datum/reagent/medicine/C2/multiver = " (Heals Toxin)", /datum/reagent/medicine/C2/aiuri = " (Heals Burn)", /datum/reagent/medicine/epinephrine = " (Stabilizes Critical Patients)", /datum/reagent/medicine/spaceacillin = " (Anti-Viral)", /datum/reagent/medi ...
```
```suggestion
	var/static/list/reagent_desc = list(/datum/reagent/medicine/C2/convermol = " (Heals Oxygen)", /datum/reagent/medicine/C2/libital = " (Heals Brute)", /datum/reagent/medicine/C2/multiver = " (Heals Toxin)", /datum/reagent/medicine/C2/aiuri = " (Heals Burn)", /datum/reagent/medicine/epinephrine = " (Stabilizes Critical Patients)", /datum/reagent/medicine/spaceacillin = " (Anti-Viral)", /datum/reagent/medicine/salglu_solution = " (Temp. Blood Replacement)")
```

Rohesie replied:
Or a global list in case you want variations.

V2LenKagamine committed:
Update code/modules/reagents/reagent_containers/borghydro.dm

List now static and looks better

Co-Authored-By: spookydonut <github@spooksoftware.com>

ExcessiveUseOfCobblestone commented on code/modules/reagents/reagent_containers/borghydro.dm:
```diff
@@ -32,7 +32,7 @@ Borg Hypospray
 								//Used as list for input() in shakers.
 	var/list/reagent_names = list()
-
+	var/static/list/reagent_desc = list(/datum/reagent/medicine/C2/convermol = " (Heals Oxygen)", /datum/reagent/medicine/C2/libital = " (Heals Brute)", /datum/reagent/medicine/C2/multiver = " (Heals Toxin)", /datum/reagent/medicine/C2/aiuri = " (Heals Burn)", /datum/reagent/medicine/epinephrine = " (Stabilizes Critical Patients)", /datum/reagent/medicine/spaceacillin = " (Anti-Viral)", /datum/reage ...
```
doc req

Rohesie commented on code/modules/reagents/reagent_containers/borghydro.dm:
```diff
@@ -1,4 +1,4 @@
-#define C2NAMEREAGENT	"[initial(reagent.name)] (Has Side-Effects)"
+#define CHEMEFFECT "[initial(reagent.name)][(reagent in reagent_desc) ? reagent_desc[reagent] : ""] [initial(reagent.harmful) ? "(Has side effects)" : ""]"
```
```suggestion
#define CHEMEFFECT "[initial(reagent.name)][reagent_desc[reagent]] [initial(reagent.harmful) ? "(Has side effects)" : ""]"
```
Same effect, if it's not defined in the associative list it will resolve to `null`, and that's just the same as `""`

V2LenKagamine committed:
Update code/modules/reagents/reagent_containers/borghydro.dm

Improves CHEMEFFECT define

Co-Authored-By: Rohesie <rohesie@gmail.com>

MarioWizard119 commented:
An Asimov borg is unable to use convermol nor synthflesh, since they technically deal toxin damage.

From the rules page on silicon policy, “1. An Asimov-compliant silicon cannot intentionally inflict harm, even if a minor amount of harm would prevent a major amount of harm. 2. Lesser immediate harm takes priority over greater future harm.”

This means conversion chems are off the table for an Asimov borg, so why the fuck borgs even have convermol, I don’t know.

LemonInTheDark commented:
What a related topic

MarioWizard119 commented:
Oh, just on the topic of potential accidental law 1 violations. Accidental human harm and whatnot.

trollbreeder commented:
> An Asimov borg is unable to use convermol nor synthflesh, since they technically deal toxin damage.

can we have tirimol instead then? it's a dual sided, EMERGENT GAMEPLAY ADDING thing which can nonharmfully help with suffocation, or be a knockout chem that can lead into a stronger sedative like chloral hydrate or morphine

ExcessiveUseOfCobblestone commented:
C2 side effects are not considered intentional harming, they are long-term side effects ergo being law 1 compliant (borgs do not\should not define harm by damage vars in the code going up).

That said, I would be fine with tirimol replacing it. They should be relatively on par with each other anyways.

ExcessiveUseOfCobblestone commented:
Re-PR once you have properly documented.
